shared/actions/chat.js

```diff
@@ -1142,19 +1146,31 @@ function * _sendNotifications (action: AppendMessages): SagaGenerator<any, any>
   }
 }
 
+function * _markThreadsStale (action: MarkThreadsStale): SagaGenerator<any, any> {
+  const selectedConversation = yield select(_selectedSelector)
+  yield put({type: 'chat:clearMessages', payload: {conversationIDKey: selectedConversation}})
+  yield put(loadMoreMessages(selectedConversation, false))
+}
+
+function _threadIsCleared (action: Action, nextAction: Action): boolean {
```

Comment:
Not sure about a better name, but this naming implies that nextAction comes after `action` which isn't necessarily the case.

Maybe something else?